File: exercise/체육복 .java
import java.util.Arrays;  

class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        Arrays.sort(lost);
        Arrays.sort(reserve);
        int answer = n;
        for (int i = 0; i < lost.length; i++){  
            boolean rent = false;
            int j = 0;
            while(rent == false){
                if(j == reserve.length){
                    break;
                }
                else if(lost[i] == reserve[j]){
                    reserve[j] = -1;
                    rent = true;
                }
                else if(lost[i] == reserve[j]+1){
                    reserve[j] = -1;
                    rent = true;
                } 
                else if(lost[i] == reserve[j]-1){
                    reserve[j] = -1;
                    rent = true;
                }
                else{
                    j++;
                }
            }
            if(rent == false){
                    answer--;
                }
        }
        return answer;
    }
}

File: exercise/체육복.py
def solution(n, lost, reserve):
    answer = n
    lost.sort()
    reserve.sort()

    for i in lost:
        if i in reserve:
            reserve.pop()
            lost.pop()
            

    # for i in reserve:
    #     if i-1 in lost:
    #         lost.remove(i-1)
    #         continue
    #     if i+1 in lost:
    #         lost.remove(i+1)
    #         continue
    return answer - len(lost)

print(solution( 2, [2, 1], [1, 2]))

